test/LRPTests/APITests.cpp

```diff
@@ -0,0 +1,167 @@
+                &processInfo
+            );
+
+            if (wasProcessCreated == FALSE)
+            {
+                VERIFY_SUCCEEDED(GetLastError());
+            }
+
+            // Wait for the process to come up and be captured in the snapshot from verification step.
+            Sleep(1000);
+            VerifyLRP_IsRunning(true);
+        }
+
+        void VerifyLRP_IsRunning(bool isRunning)
+        {
+            wil::unique_handle processesSnapshot(CreateToolhelp32Snapshot(TH32CS_SNAPPROCESS, 0));
+            VERIFY_IS_NOT_NULL(processesSnapshot.get());
+
+            PROCESSENTRY32 processEntry = { 0 };
+            processEntry.dwSize = sizeof(processEntry);
+
+            BOOL result = Process32First(processesSnapshot.get(), &processEntry);
+            while (result != FALSE)
+            {
+                if (wcscmp(L"PushNotificationsLongRunningTask.exe", processEntry.szExeFile) == 0)
+                {
+                    VERIFY_IS_TRUE(isRunning);
+                    DWORD processId = processEntry.th32ProcessID;
+
+                    wil::unique_handle longRunningProcessHandle(OpenProcess(PROCESS_QUERY_INFORMATION, FALSE, processId));
+                    DWORD exitCode = 0;
+                    BOOL exitCodeProcess = GetExitCodeProcess(longRunningProcessHandle.get(), &exitCode);
+
+                    VERIFY_SUCCEEDED(exitCodeProcess == FALSE ? GetLastError() : S_OK);
+                    VERIFY_ARE_EQUAL(exitCode, STILL_ACTIVE);
+
+                    return;
+                }
+
+                result = Process32Next(processesSnapshot.get(), &processEntry);
+            }
+
+            VERIFY_IS_FALSE(isRunning);
+        }
+
+        TEST_METHOD(RegisterUnregisterLongRunningActivator)
+        {
+            winrt::com_ptr notificationPlatform{ winrt::try_create_instance<INotificationsLongRunningPlatform>(_uuidof(NotificationsLongRunningPlatform), CLSCTX_ALL) };
+            VERIFY_IS_NOT_NULL(notificationPlatform.get());
+
+            VERIFY_NO_THROW(notificationPlatform->RegisterLongRunningActivator(c_processName));
+            VERIFY_NO_THROW(notificationPlatform->UnregisterLongRunningActivator(c_processName));
+        }
+
+        TEST_METHOD(RegisterUnregisterLongRunningActivatorWithClsid)
+        {
+            winrt::com_ptr notificationPlatform{ winrt::try_create_instance<INotificationsLongRunningPlatform>(_uuidof(NotificationsLongRunningPlatform), CLSCTX_ALL) };
+            VERIFY_IS_NOT_NULL(notificationPlatform.get());
+
+            VERIFY_NO_THROW(notificationPlatform->RegisterLongRunningActivatorWithClsid(c_processName, c_remoteId));
+            VERIFY_NO_THROW(notificationPlatform->UnregisterLongRunningActivator(c_processName));
+        }
+
+        TEST_METHOD(AddToastRegistrationMappingNoSink)
+        {
+            winrt::com_ptr notificationPlatform{ winrt::try_create_instance<INotificationsLongRunningPlatform>(_uuidof(NotificationsLongRunningPlatform), CLSCTX_ALL) };
```

Comment:
As long as a client has a reference to the notificationPlatform, the LRP will stay alive. We don't want this behavior for LaunchLRP_FromStartupTask since we are verifying the LRP is dead then comes alive in the test from the startuptask.exe. If it wasn't for this test, we could use a member variable.